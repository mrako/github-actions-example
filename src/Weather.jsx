import React, { Component } from 'react';

const baseURL = process.env.ENDPOINT;

const getWeatherFromApi = async () => {
  try {
    const response = await fetch(`${baseURL}/weather`);
    return response.json();
  } catch (error) {
    console.error(error); // eslint-disable-line no-console
  }

  return {};
};

class Weather extends Component {
  constructor(props) {
    super(props);

    this.state = {
      icon: '',
    };
  }

  async componentDidMount() {
    const weather = await getWeatherFromApi();

    if (weather.icon) {
      this.setState({ icon: weather.icon.slice(0, -1) });
    }
  }

  render() {
    const { icon } = this.state;

    return (
      <div className="icon">
        { icon && <img alt={icon} src={`/img/${icon}.svg`} /> }
      </div>
    );
  }
}

export default Weather;
