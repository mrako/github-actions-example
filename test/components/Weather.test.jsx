import React from 'react';
import renderer from 'react-test-renderer';

import Weather from '../../src/Weather';

const mockFetchPromise = Promise.resolve({ json: () => Promise.resolve({}) });

global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

describe('Weather', () => {
  it('renders correctly', () => {
    const component = (
      <Weather />
    );
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
