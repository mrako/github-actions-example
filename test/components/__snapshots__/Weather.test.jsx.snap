// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather renders correctly 1`] = `
<div
  className="icon"
>
  
</div>
`;
